src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -195,52 +199,119 @@ StaticLLMPipeline::StaticLLMPipeline(
+        const auto kMinResponseLen = pop_or_default(pipeline_config, "MIN_RESPONSE_LEN", 150u);
+        // FIXME For some models KV-cache dim != 2u
+        m_kvcache_desc = KVCacheDesc { kMaxPromptLen, kMaxPromptLen + kMinResponseLen, 0u, 2u };
+        reshape_to_static(m_prefill_model, m_kvcache_desc.max_prompt_size, m_kvcache_desc.max_prompt_size);
+        reshape_to_static(m_kvcache_model, 1u, m_kvcache_desc.total_size);
+        // (7) Compile both model
+        // FIXME: Drop CACHE_DIR option if NPUW is enabled
+        merge_config_with(prefill_config, pipeline_config);
+        merge_config_with(generate_config, pipeline_config);
+        drop_cache_dir(prefill_config);
+        drop_cache_dir(generate_config);
+        m_prefill_request = core.compile_model(
+            m_prefill_model, device, prefill_config
+        ).create_infer_request();
+        m_kvcache_request = core.compile_model(
+            m_kvcache_model, device, generate_config
+        ).create_infer_request();
+    } else {
+        auto import_blob = [this, 
+                            &path,
+                            &pipeline_config,
+                            &core,
+                            &device](const std::string& model_name,
+                                     ov::AnyMap& model_config) {
+            auto blob_path = pop_or_default(model_config, "BLOB_PATH", std::string{});
+
+            if (blob_path.empty()) {
+                blob_path = (path /
+                    (std::string("openvino_") + model_name + ".blob")).generic_string();
+            }
+
+            if (!std::filesystem::exists(blob_path)) {
+                OPENVINO_THROW("Blob for " + model_name + " model is not found at: "
+                    + blob_path);
+            }
+
+            merge_config_with(model_config, pipeline_config);
+
+            std::fstream fs(blob_path, std::ios::in | std::ios::binary);
+
+            return core.import_model(
+                fs, device, model_config);
+
+        };
+
+        auto get_kvcache_size = [](ov::CompiledModel& model) {
+            for (auto input : model.inputs()) {
+                const auto& input_name = input.get_any_name();
+                if (input_name.find("attention_mask") != std::string::npos) {
+                    return input.get_shape()[1];
+                }
+            }
+            return uint64_t(0);
+        };
+
+        // (1) Import prefill model from model directory or specified path
+        auto prefill_model = import_blob("prefill", prefill_config);
+        m_prefill_request = prefill_model.create_infer_request();
+        // (2) Import generate model from model directory or specified path
+        auto generate_model = import_blob("generate", generate_config);
+        m_kvcache_request = generate_model.create_infer_request();
+        // (3) Fill in m_kvcache_desc
+        const uint32_t kMaxPromptLen = uint32_t(get_kvcache_size(prefill_model));
```

Comment:
Maybe we should prohibit to specify `MAX_PROMPT_LEN` & `MIN_RESPONSE_LEN` in case of blobs

Reply:
Fixed, thanks!